vault/ui_custom_messages/entry_test.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package uicustommessages
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestEntryFindMessages verifies that the (*Entry).FindMessages method behaves
+// correctly in the different edge cases that could occur.
+func TestEntryFindMessages(t *testing.T) {
+	testMessagesMap := map[string]Message{
+		"post-login": {
+			Authenticated: true,
+		},
+		"modal": {
+			Type: "modal",
+		},
+		"active": {
+			StartTime: time.Now().Add(-1 * time.Hour),
+			EndTime:   time.Now().Add(time.Hour),
+		},
+	}
+
+	for _, testcase := range []struct {
+		name             string
+		entry            Entry
+		filter           FindFilter
+		expectResultsLen int
+	}{
+		{
+			name:             "messages map is nil: filter empty",
+			entry:            Entry{},
+			filter:           FindFilter{},
+			expectResultsLen: 0,
+		},
+		{
+			name: "filter matches",
+			entry: Entry{
+				Messages: testMessagesMap,
+			},
+			filter: FindFilter{
+				messageType: "modal",
+			},
+			expectResultsLen: 1,
+		},
+		{
+			name: "filter does not match",
+			entry: Entry{
+				Messages: testMessagesMap,
+			},
+			filter: FindFilter{
+				messageType: "banner",
+			},
+			expectResultsLen: 0,
+		},
+	} {
+		result := testcase.entry.FindMessages(testcase.filter)
+
+		assert.NotNil(t, result, testcase.name)
+		assert.Equal(t, testcase.expectResultsLen, len(result))
+	}
+}
+
+// TestEntryCreateMessage verifies that the (*Entry).CreateMessage method
+// behaves correctly in the different edge cases that could occur.
+func TestEntryCreateMessage(t *testing.T) {
```

Comment:
I said it before on my previous review but it's worth restating: I love the tests you've made for this feature!